compiler/rustc_parse/src/lexer/mod.rs

```diff
@@ -553,6 +544,58 @@ impl<'a> StringReader<'a> {
         err.emit()
     }
 
+    fn report_unterminated_block_comment(&self, start: BytePos, doc_style: Option<DocStyle>) {
+        let msg = match doc_style {
+            Some(_) => "unterminated block doc-comment",
+            None => "unterminated block comment",
+        };
+        let last_bpos = self.pos;
+        let mut err = self.sess.span_diagnostic.struct_span_fatal_with_code(
+            self.mk_sp(start, last_bpos),
+            msg,
+            error_code!(E0758),
+        );
+        let mut nested_block_comment_open_idxs = vec![];
+        let mut last_nested_block_comment_idxs = None;
+        let mut content_chars = self.str_from(start).char_indices();
+
+        if let Some((_, mut last_char)) = content_chars.next() {
+            while let Some((idx, c)) = content_chars.next() {
```

Comment:
Maybe using zip or windows(2) will be much more readable, but I didn't found any usage of both under rustc_parse so not sure if it's ok to use them.

Reply:
It's ok to use any standard library features in this crate if they improve the code.

Reply:
I decided to change to use `peekable` instead as `itertools` is not standard library and zip seems need clone 1b7008dc77e25049b04e5c3e31aecf4de00803e7